Release 2021.2.0 for MKL and 2021.2.2 for DAL

antkneed opened the pull request:
- MKL 2021 run constrains on GLIBC and OSX.
- Do not re-package `dal` on windows `win-32` as it is not provided upstream.
- License standardization between unix & windows as they are located in the same place now.
- `daal` -> `dal` updates.

Commit:
Update for MKL 2021.2.0 & DAL 2021.2.2

Commit:
Adding note around mkl-service export for mkl-devel.

Commit:
Cleaning up recipe comments.

napetrov commented on recipe/meta.yaml:
```diff
@@ -160,20 +204,23 @@ outputs:
+      number: {{ daal_buildnum|int + dstbuildnum|int }}
       binary_relocation: false
       detect_binary_files_with_prefix: false
+      skip: True                                  # [win32]
```
It's true not only for windows but for linux. 32 bit support have been dropped in 2021.* versions

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+        - tbb {{ daal_version.split('.')[0] }}.*
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: DAL runtime libraries
```
```suggestion
      summary: Intel® oneDAL runtime libraries
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      skip: True                                  # [win32]
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Headers for building against DAL libraries
```
```suggestion
      summary: Headers for building against Intel® oneDAL libraries
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+        - tbb {{ daal_version.split('.')[0] }}.*
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Static libraries for DAL
```
```suggestion
      summary: Static libraries for Intel® oneDAL
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+        - {{ pin_subpackage('dal', exact=True) }}
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Devel package for building things linked against DAL shared libraries
```
```suggestion
      summary: Devel package for building things linked against Intel® oneDAL shared libraries
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      summary: DAL runtime libraries
+      license: Intel Simplified Software License
+      license_family: Proprietary
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal/info/licenses/license.txt
         - dal/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      license_file: mkl/info/licenses/license.txt
+      license_url: https://software.intel.com/content/www/us/en/develop/articles/end-user-license-agreement.html
+      doc_url: https://software.intel.com/content/www/us/en/develop/tools.html
+      dev_url: https://software.intel.com/content/www/us/en/develop/tools.html
```
```suggestion
      dev_url: https://github.com/oneapi-src/oneDAL
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      license_file: mkl/info/licenses/license.txt
+      license_url: https://software.intel.com/content/www/us/en/develop/articles/end-user-license-agreement.html
+      doc_url: https://software.intel.com/content/www/us/en/develop/tools.html
+      dev_url: https://software.intel.com/content/www/us/en/develop/tools.html
```
```suggestion
      dev_url: https://github.com/oneapi-src/oneDAL
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      license_file: mkl/info/licenses/license.txt
+      license_url: https://software.intel.com/content/www/us/en/develop/articles/end-user-license-agreement.html
+      doc_url: https://software.intel.com/content/www/us/en/develop/tools.html
+      dev_url: https://software.intel.com/content/www/us/en/develop/tools.html
```
```suggestion
      dev_url: https://github.com/oneapi-src/oneDAL
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      license_file: mkl/info/licenses/license.txt
+      license_url: https://software.intel.com/content/www/us/en/develop/articles/end-user-license-agreement.html
+      doc_url: https://software.intel.com/content/www/us/en/develop/tools.html
+      dev_url: https://software.intel.com/content/www/us/en/develop/tools.html
```
```suggestion
      dev_url: https://github.com/oneapi-src/oneDAL
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      summary: Headers for building against DAL libraries
+      license: Intel Simplified Software License
+      license_family: Proprietary
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-include/info/licenses/license.txt
         - dal-include/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      summary: Static libraries for DAL
+      license: Intel Simplified Software License
+      license_family: Proprietary
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-static/info/licenses/license.txt
         - dal-static/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+      summary: Devel package for building things linked against DAL shared libraries
+      license: Intel Simplified Software License
+      license_family: Proprietary
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-devel/info/licenses/license.txt
         - dal-devel/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
       license_family: Proprietary
-      license_file: mkl/info/LICENSE.txt  # [win]
-      license_file: mkl/info/licenses/license.txt  # [unix]
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-static/info/licenses/license.txt
         - dal-static/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
       license_family: Proprietary
-      license_file: mkl/info/LICENSE.txt  # [win]
-      license_file: mkl/info/licenses/license.txt  # [unix]
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-include/info/licenses/license.txt
         - dal-include/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
       license_family: Proprietary
-      license_file: mkl/info/LICENSE.txt  # [win]
-      license_file: mkl/info/licenses/license.txt  # [unix]
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal/info/licenses/license.txt
         - dal/info/licenses/tpp.txt
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: DAL runtime libraries
+      license: Intel Simplified Software License
```
```suggestion
      description: |
        Intel® Data Analytics Acceleration Library (Intel® oneDAL) is the library of Intel® architecture
        optimized building blocks covering all stages of data analytics: data acquisition from a data
        source, preprocessing, transformation, data mining, modeling, validation, and decision making.
        This package is a repackaged set of binaries obtained directly from Intel\'s anaconda.org channel.
      license: Intel Simplified Software License
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Headers for building against DAL libraries
+      license: Intel Simplified Software License
```
```suggestion
      description: |
        Intel® Data Analytics Acceleration Library (Intel® oneDAL) is the library of Intel® architecture
        optimized building blocks covering all stages of data analytics: data acquisition from a data
        source, preprocessing, transformation, data mining, modeling, validation, and decision making.
        This package is a repackaged set of binaries obtained directly from Intel\'s anaconda.org channel.
      license: Intel Simplified Software License
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Static libraries for DAL
+      license: Intel Simplified Software License
```
```suggestion
      description: |
        Intel® Data Analytics Acceleration Library (Intel® oneDAL) is the library of Intel® architecture
        optimized building blocks covering all stages of data analytics: data acquisition from a data
        source, preprocessing, transformation, data mining, modeling, validation, and decision making.
        This package is a repackaged set of binaries obtained directly from Intel\'s anaconda.org channel.
      license: Intel Simplified Software License
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
+    about:
+      home: https://software.intel.com/content/www/us/en/develop/tools.html
+      summary: Devel package for building things linked against DAL shared libraries
+      license: Intel Simplified Software License
```
```suggestion
      description: |
        Intel® Data Analytics Acceleration Library (Intel® oneDAL) is the library of Intel® architecture
        optimized building blocks covering all stages of data analytics: data acquisition from a data
        source, preprocessing, transformation, data mining, modeling, validation, and decision making.
        This package is a repackaged set of binaries obtained directly from Intel\'s anaconda.org channel.
      license: Intel Simplified Software License
```

napetrov commented on recipe/meta.yaml:
```diff
@@ -200,80 +254,170 @@ outputs:
       license_family: Proprietary
-      license_file: mkl/info/LICENSE.txt  # [win]
-      license_file: mkl/info/licenses/license.txt  # [unix]
+      license_file: mkl/info/licenses/license.txt
```
```suggestion
      license_file:
         - dal-devel/info/licenses/license.txt
         - dal-devel/info/licenses/tpp.txt
```

antkneed committed:
Apply suggestions from code review

Leveraging the suggestions as a starting place for the license and description updates. Clean-up and testing to follow.

Co-authored-by: Nikolay Petrov <nikolay.a.petrov@intel.com>

Commit:
License files are included via the meta.yaml directly.

Commit:
Updating license path for dal-devel.
